factory/faker.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, provider, locale=None, **kwargs):
         self.provider_kwargs = kwargs
         self.locale = locale
 
-    def generate(self, extra_kwargs):
+    def generate(self, extra_kwargs={}):
```

Comment:
Almost never a good idea to use `{}` as a default value: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
ah yes, rookie mistake, thanks for spotting it.